src/webpub_manifest_parser/core/ast.py

```diff
@@ -0,0 +1,831 @@
+
+    clipped = BooleanProperty("clipped", False)
+    continuous = BooleanProperty("continuous", False)
+    fit = EnumProperty("fit", False, ["width", "height", "contain", "cover"])
+    orientation = EnumProperty("orientation", False, ["auto", "landscape", "portrait"])
+    overflow = EnumProperty(
+        "overflow", False, ["auto", "paginated", "scrolled", "scrolled-continuous"]
+    )
+    spread = EnumProperty("spread", False, ["auto", "both", "none", "landscape"])
+
+
+class CompactCollection(Node):
+    """A compact collection is defined as a grouping of links."""
+
+    links = ArrayOfLinksProperty(key="links", required=True)
+
+    def __init__(self, role=None, links=None):
+        """Initialize a new instance of Collection class.
+
+        :param role: Collection's role (can be empty when self is a manifest)
+        :type role: Optional[CollectionRole]
+
+        :param links: Collection's links
+        :type links: Optional[LinksList]
+        """
+        super(CompactCollection, self).__init__()
+
+        self._role = role
+        self.links = links
+
+    @property
+    def role(self):
+        """Return the collection's role.
+
+        :return: Collection's role.
+        :rtype: Optional[CollectionRole]
+        """
+        return self._role
+
+
+class Collection(CompactCollection):
+    """A collection is defined as a grouping of metadata, links and sub-collections."""
+
+    def __init__(self, role=None, links=None, metadata=None):
+        """Initialize a new instance of Collection class.
+
+        :param role: Collection's role (can be empty when self is a manifest)
+        :type role: Optional[CollectionRole]
+
+        :param links: Collection's links
+        :type links: Optional[LinksList]
+
+        :param metadata: Collection's metadata
+        :type metadata: Optional[Metadata]
+        """
+        super(Collection, self).__init__(role, links)
+
+        self._role = role
+        self._sub_collections = CollectionList()
+        self.metadata = metadata
+
+    metadata = TypeProperty(
+        key="metadata", required=True, nested_type=PresentationMetadata
+    )
+
+    @property
+    def sub_collections(self):
+        """Return a list of sub-collections.
+
+        :return: List of sub-collections.
+        :rtype: CollectionList
+        """
+        return self._sub_collections
+
+    @property
+    def compact(self):
+        """Return a boolean value indicating if this collection is compact.
+
+        :return: Boolean value indicating if this collection is compact
+        :rtype: bool
+        """
+        return self.metadata is None and len(self._sub_collections) == 0
+
+    @property
+    def full(self):
+        """Return a boolean value indicating if this collection is full.
+
+        :return: Boolean value indicating if this collection is full
+        :rtype: bool
+        """
+        return self.metadata is not None and len(self._sub_collections) > 0
```

Comment:
In practice I don't know any case where knowing if a collection is compact vs full is important. Did you need to use it?

Reply:
I'm the original author of this and I think I only used it in enforcing the rules laid down by the spec (e.g. `links` must be a compact collection).

Reply:
Ha yes in that case that makes sense. We used simple `Link` arrays for `readingOrder` and such on mobile.